packages/cubejs-schema-compiler/src/adapter/PreAggregations.js

```diff
@@ -362,82 +363,175 @@ export class PreAggregations {
     }
   }
 
+  /**
+   * Returns function to determine whether pre-aggregation can be used or not
+   * for specified query, or its value for `refs` if specified.
+   * @param {Object} transformedQuery transformed query
+   * @param {Object?} refs pre-aggs reference
+   * @returns {function(preagg: Object): boolean}
+   */
   static canUsePreAggregationForTransformedQueryFn(transformedQuery, refs) {
+    /**
+     * Returns an array of 2-elements arrays with the dimension and granularity
+     * sorted by the concatenated dimension + granularity key.
+     * @param {Array<{dimension: string, granularity: string}>} timeDimensions
+     * @returns {Array<Array<string>>}
+     */
+    const sortTimeDimensions = (timeDimensions) => (
+      timeDimensions &&
+      R.sortBy(
+        d => d.join('.'),
+        timeDimensions.map(
+          d => [
+            d.dimension,
+            d.granularity || 'day', // TODO granularity shouldn't be null?
+          ]
+        ),
+      ) || []
+    );
+
+    /**
+     * @type {Set<string>}
+     */
     const filterDimensionsSingleValueEqual =
       transformedQuery.filterDimensionsSingleValueEqual instanceof Set
         ? transformedQuery.filterDimensionsSingleValueEqual
         : new Set(
-          Object.keys(transformedQuery.filterDimensionsSingleValueEqual || {})
+          Object.keys(
+            transformedQuery.filterDimensionsSingleValueEqual || {},
+          )
         );
+
+    /**
+     * Determine whether pre-aggregation can be used or not.
```

Comment:
This comment could be better by stating the criteria that make a pre-aggregation usable or not. That could help establishing intent, and identifying bugs if the code diverges from the stated intent.

Reply:
Well, this is a good point. To be honest, I don't have a better comment for this function right now because I don't know all criteria we have. I need to invest a lot of time in the research to understand it. And as always, I don't have it ))